articles/quickstart/spa/vanillajs-beta/_includes/_centralized_login.md

```diff
@@ -0,0 +1,374 @@
+      window.history.replaceState({}, document.title, "/");
+    }
+  }
+};
+```
+
+Now the redirect is properly handled and the authentication can be completed successfully, but you need to remove the query parameters from the URL so that if the user refreshes the page this is not considered as if a new exchange needs to be done. This is achieved with the `window.history.replaceState` method.
+
+> **Checkpoint:** Run the project and click the "Log in" button. You should be taken to the Universal Login Page configured for your application. Go ahead and create a new user or log in using a social connection. After authenticating successfully, you will be redirected to the page you were before. This time, the result will be present in the URL query and the exchange will happen automatically. If everything went fine, you will end up with no query parameters in the URL, the user would now be logged in and the "Log out" button will be enabled.
+
+If at this part you see any errors on the Auth0 page, check that you have not forgotten to whitelist the callback URL or the allowed origins as explained initially.
+
+### Log the user out
+
+You may have noticed that the **Log out** button is clickable when the user is authenticated, but does nothing. You need to add the code that will log the user out from the Auth0 backend.
+
+Start the log out by calling the `auth0.logout()` method passing a valid return-to URI. In this sample you will return the user back to the same page they are now. You can obtain that value from `window.location.origin` property. Abstract this logic into a `logout()` method.
+
+```js
+// public/js/app.js
+
+const logout = () => {
+  auth0.logout({
+    returnTo: window.location.origin
+  });
+};
+```
+
+> **Checkpoint:** Being authenticated click the **Log out** button. You should be taken to the Universal Login Page configured for your application and then back to the page you were before. Now the authentication cookies were cleared and the user is logged out. The "Log in" button will be enabled back again.
+
+If at this part you see any errors on the Auth0 page, check that you have not forgotten to whitelist the logout url as explained initially.
+
+### Read the user profile
+
+Everytime a user is logged in you get access both to the **access token** and the associated **user profile** information. Typically, the token is used to call your backend application and the profile information to display their name and profile picture. In this guide you are going to display them in separate text areas so you can easily inspect them. Open the `index.html` file and insert the following lines at the bottom of the body.
+
+```html
+<!-- index.html -->
+
+<body>
+  <!-- ... -->
+
+  <div class="hidden" id="gated-content">
+    <p>
+      You're seeing this content because you're currently
+      <strong>logged in</strong>.
+    </p>
+    <label>
+      Access token:
+      <textarea readonly="true" rows="1" id="ipt-access-token"></textarea>
```

Comment:
I like the `<pre>` block used in the calling API section a little better than this ... looks a little cleaner over. IMHO.

Reply:
Agreed